README.md

```diff
@@ -73,42 +101,87 @@ Or download [the latest `before-after-hook.min.js`](https://github.com/gr2m/befo
 
 ## API
 
-- [Constructor](#constructor)
-- [hook.api](#hookapi)
-- [hook()](#hook)
-- [hook.before()](#hookbefore)
-- [hook.error()](#hookerror)
-- [hook.after()](#hookafter)
-- [hook.wrap()](#hookwrap)
-- [hook.remove()](#hookremove)
+- [Singular Hook Constructor](#singular-hook-api)
+- [Hook Collection Constructor](#hook-collection-api)
+
+## Singular hook API
+
+- [Singular constructor](#singular-constructor)
+- [hook.api](#singular-api)
+- [hook()](#singular-api)
+- [hook.before()](#singular-api)
+- [hook.error()](#singular-api)
+- [hook.after()](#singular-api)
+- [hook.wrap()](#singular-api)
+- [hook.remove()](#singular-api)
+
+### Singular constructor
+
+The `Hook.Singular` constructor has no options and returns a `hook` instance with the
+methods below:
+
+```js
+const hook = new Hook.Singular()
+```
+Using the singular hook is recommended for [TypeScript](#typescript)
+
+### Singular API
+
+The singular hook is a reference to a single hook. This means that there's no need to pass along any identifier (such as a `name` as can be seen in the [Hook.Collection API](#collectionapi)).
+
+The API of a singular hook is exactly the same as a collection hook and we therefore suggest you read the [Hook.Collection API](#collectionapi) and leave out any use of the `name` argument. Just skip it like described in this example:
+```js
+const hook = new Hook.Singular()
+
+// good
+hook.before(beforeHook)
+hook.after(afterHook)
+hook(options, fetchFromDatabase)
 
-### Constructor
+// bad
+hook.before('get', beforeHook)
+hook.after('get', afterHook)
+hook('get', options, fetchFromDatabase)
+```
+
+## Hook collection API
 
-The `Hook` constructor has no options and returns a `hook` instance with the
+- [Collection constructor](#collection-constructor)
+- [collection.api](#collectionapi)
+- [collection()](#collection)
+- [collection.before()](#collectionbefore)
+- [collection.error()](#collectionerror)
+- [collection.after()](#collectionafter)
+- [collection.wrap()](#collectionwrap)
+- [collection.remove()](#collectionremove)
+
+### Collection constructor
+
+The `Hook.Collection` constructor has no options and returns a `hookCollection` instance with the
 methods below
 
 ```js
-const hook = new Hook()
+const collection = new Hook.Collection()
```

Comment:
can we use `hookCollection` as variable name throughout the README to be consistent? I prefer long names, clarity is kind 👑

Reply:
I completely agree. I change it to `hookCollection`